pkg/catalog/routes_test.go

```diff
@@ -24,6 +25,290 @@ import (
 	"github.com/openservicemesh/osm/pkg/utils"
 )
 
+func TestListTrafficPoliciesForTrafficSplits(t *testing.T) {
+	assert := tassert.New(t)
+
+	testSplit1 := split.TrafficSplit{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "bar",
+		},
+		Spec: split.TrafficSplitSpec{
+			Service: "apex-split-1",
+			Backends: []split.TrafficSplitBackend{
+				{
+					Service: tests.BookstoreV1ServiceName,
+					Weight:  tests.Weight10,
+				},
+				{
+					Service: tests.BookstoreV2ServiceName,
+					Weight:  tests.Weight90,
+				},
+			},
+		},
+	}
+
+	testSplit1NamespacedHostnames := []string{
+		"apex-split-1.bar",
+		"apex-split-1.bar.svc",
+		"apex-split-1.bar.svc.cluster",
+		"apex-split-1.bar.svc.cluster.local",
+		"apex-split-1.bar:8888",
+		"apex-split-1.bar.svc:8888",
+		"apex-split-1.bar.svc.cluster:8888",
+		"apex-split-1.bar.svc.cluster.local:8888",
+	}
+
+	testSplit2 := split.TrafficSplit{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "bar",
+		},
+		Spec: split.TrafficSplitSpec{
+			Service: "apex-split-1",
+			Backends: []split.TrafficSplitBackend{
+				{
+					Service: tests.BookstoreV1ServiceName,
+					Weight:  tests.Weight90,
+				},
+				{
+					Service: tests.BookstoreV2ServiceName,
+					Weight:  tests.Weight10,
+				},
+			},
+		},
+	}
+
+	testSplit3 := split.TrafficSplit{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "baz",
+		},
+		Spec: split.TrafficSplitSpec{
+			Service: "apex-split-1",
+			Backends: []split.TrafficSplitBackend{
+				{
+					Service: tests.BookstoreV1ServiceName,
+					Weight:  tests.Weight10,
+				},
+				{
+					Service: tests.BookstoreV2ServiceName,
+					Weight:  tests.Weight90,
+				},
+			},
+		},
+	}
+
+	testSplit3NamespacedHostnames := []string{
+		"apex-split-1.baz",
+		"apex-split-1.baz.svc",
+		"apex-split-1.baz.svc.cluster",
+		"apex-split-1.baz.svc.cluster.local",
+		"apex-split-1.baz:8888",
+		"apex-split-1.baz.svc:8888",
+		"apex-split-1.baz.svc.cluster:8888",
+		"apex-split-1.baz.svc.cluster.local:8888",
+	}
+
+	testCases := []struct {
+		name             string
+		sourceNamespace  string
+		trafficsplits    []*split.TrafficSplit
+		expectedPolicies []*trafficpolicy.OutboundTrafficPolicy
+		expectedRoutes   []*trafficpolicy.RouteWeightedClusters
+		apexMeshServices []service.MeshService
+	}{
+		{
+			name:            "single traffic split policy in different namespace",
+			sourceNamespace: "foo",
+			trafficsplits:   []*split.TrafficSplit{&tests.TrafficSplit},
+			apexMeshServices: []service.MeshService{
+				{
+					Name:      tests.BookstoreApexServiceName,
+					Namespace: tests.Namespace,
+				},
+			},
+			expectedPolicies: []*trafficpolicy.OutboundTrafficPolicy{
+				{
+					Name:      "bookstore-apex-default",
```

Comment:
You will need to fix the name here and in all other test cases
```suggestion
					Name:      "bookstore-apex.default",
```

Reply:
updated. thank you